spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -998,8 +1015,8 @@ contract PerpetualTranche is
     /// @dev Queries the current subscription state of the perp and vault systems.
     function _querySubscriptionState() private view returns (IFeePolicy.SubscriptionParams memory s) {
         s.perpTVL = _reserveValue();
-        s.vaultTVL = IVault(vault).getTVL();
-        (s.perpTR, s.vaultTR) = _depositBond.getSeniorJuniorRatios();
+        s.vaultTVL = (address(vault) != address(0)) ? IVault(vault).getTVL() : 0;
```

Comment:
Do we ever expect vault to be 0x0? Worth the gas for this check?

A lot of things might break, like anything dependent on SRs. Seems like we should write the entire system to work with that case, or not worry about it.

We could enforce during updateVault that it's != 0x0

Reply:
The `onlyVault` modifier allows all callers to rollover when the vault address (is not set) or set to address(0). However with this current version, the fee policy expects a valid vault reference to compute rollover fees and thus a vault reference has to be set.

I agree, here we can get rid of the 0x0 check. Also in the `onlyVault` modifier, as well.